framework/validators/UniqueValidator.php

```diff
@@ -190,8 +196,11 @@ private function modelExists($targetClass, $conditions, $model)
     private function prepareQuery($targetClass, $conditions)
     {
         $query = $targetClass::find();
-        $query->andWhere($conditions);
-
+        if ($this->combineType === 'or') {
+            $query->orWhere($conditions);
```

Comment:
This is unsafe: it will use `OR` for any existing condition, which may be already placed at query, default scope - in particular.
Should be always `andWhere()`, while conditions should be composed in array:

```php
[
    'and', // or 'or'
    $condition1,
    $condition2,
    ...
]
```